components/nexus-httpclient/src/main/java/org/sonatype/nexus/httpclient/config/NonProxyHostsValidator.java

```diff
@@ -26,6 +27,8 @@
 public class NonProxyHostsValidator
     extends ConstraintValidatorSupport<NonProxyHosts, String[]>
 {
+  private static final Pattern NON_PROXY_HOST_PATTERN = Pattern.compile("^((\\p{IsAlphabetic}|\\d|\\-|\\_|\\*)+\\.?)*$", Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
I am leaning to use `InetAddress.getByName` instead now, as validating possible inputs (hostname, IP4, IP6) is getting out of hand with regexp. Only possible drawback is DNS lookup in case of hostname (in case of IP addresses, it will do exactly what we want: check address format.

WDYT?